### PR TITLE
fix(common): consolidate npm publishing

### LINKED ISSUE #7310: chore(developer): Deprecate `@keymanapp/lexical-model-compiler` and publish `@keymanapp/kmc*` modules on npm when we merge feature-ldml 🙀
TODO: Deprecate `@keymanapp/lexical-model-compiler` and publish `@keymanapp/kmc*` modules on npm for v16 onward.

_Originally posted by @mcdurdin in https://github.com/keymanapp/keyman/issues/7299#issuecomment-1250135159_

NOTE: the publish script has `DRY_RUN` force-set which needs to be reset once we are ready to go.

### LINKED ISSUE #8586: chore(common): common/models/types uses old npm publish code
This should be updated to a builder script and use builder_publish_to_npm instead.

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #8586.
Fixes #7310.

Four fixes all centred around npm publish:
* Sets dependency versions when publishing npm packages
* All npm publish actions in the repository are now run as part of the Developer release build, hosted at present by kmc, until we setup a higher-level build script to do it.
* Adds a 'pack' action which mirrors 'publish' for all npm packages.
* Removes unnecessary dev dependencies of models-templates and models-wordbreakers from kmc-model.

The Web Release build CI step has been updated to check for presence of .build-builder for models-types.

@keymanapp-test-bot skip

### CODE REVIEW

File: common/models/types/build.sh [hunk 1]
```diff
@@ -10,101 +10,63 @@ set -eu
...
+. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+
+
+builder_describe "Build Keyman kmc Package Compiler module" \
+  "@/common/web/keyman-version" \
+  "clean" \
+  "configure" \
+  "build" \
+  "test" \
+  "pack" \
```
Comment (darcywong00) [inline common/models/types/build.sh:26]: Can we have a description of pack?  ...mirrors 'publish' for all npm packages

Comment (jahorton) [reply common/models/types/build.sh:26]: Huh, and it usually comes second here, after `publish`... yeah, could definitely use some more clarity there on exactly what the distinction is.

Comment (mcdurdin) [reply common/models/types/build.sh:26]: "build a local .tgz pack for testing" okay? See #8588, will fix up other refs shortly.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.87-alpha](https://keyman.com/downloads/releases/alpha/17.0.87)

**Pull request merged**